flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SchedulerTestingUtils.java

```diff
@@ -224,16 +224,21 @@ public static void canceledExecution(
     public static void setExecutionToRunning(
             DefaultScheduler scheduler, JobVertexID jvid, int subtask) {
         final ExecutionAttemptID attemptID = getAttemptId(scheduler, jvid, subtask);
+        scheduler.updateTaskExecutionState(
```

Comment:
Wouldn't it be better to have a separate `setExecutionToRecovering`? We do not switch over multiple states in any of the other methods and it might be surprising for future users of this class.